gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/index/GfxdIndexManager.java

```diff
@@ -1044,7 +1048,10 @@ public void postEvent(LocalRegion owner, EntryEventImpl event,
     }
     final Operation op = event.getOperation();
     //try {
-    final TXStateInterface tx = event.getTXState(owner);
+    TXStateInterface tx = event.getTXState(owner);
+    if (tx != null && tx.isSnapshot()) {
+      tx = null;
+    }
```

Comment:
As I had mentioned elsewhere, a separate method in event will be better (that internally checks and returns null if its snapshot).